### PR TITLE
manifest: add utilities for encrypted SeedShares

### PR DESCRIPTION
**Pull request by burgerdev**
This centralizes the algorithms used for managing seed shares and owner keys.

### CODE REVIEW

**Comment by katexochen** [comment]
I don't like how this change puts business logic into a pure crypto library. Imo the package shouldn't be aware of seed shares or import the api packages. Maybe we can move this somewhere else?

**Comment by burgerdev** [comment]
My first pick would have been the `userapi` package, but you can make a similar argument there (it's a pure transport layer package). Manifest sounds good.

**Comment by burgerdev** [comment]
Fixed, ptal.

File: internal/manifest/crypto.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+	"github.com/edgelesssys/contrast/internal/userapi"
+)
+
+// MarshalSeedShareOwnerKey converts a public key into the format for userapi.SetManifestRequest.
+func MarshalSeedShareOwnerKey(pubKey *rsa.PublicKey) HexString {
+	return NewHexString(x509.MarshalPKCS1PublicKey(pubKey))
+}
+
+// ParseSeedShareOwnerKey reads a public key embedded in a userapi.SetManifestRequest.
+func ParseSeedShareOwnerKey(pubKeyHex HexString) (*rsa.PublicKey, error) {
```
Comment (katexochen) [inline internal/manifest/crypto.go:22]: UnmarshalSeed... ?

File: internal/manifest/crypto_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"testing"
+
+	"github.com/edgelesssys/contrast/internal/userapi"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncryptDecryptSingleKey(t *testing.T) {
```
Comment (katexochen) [inline internal/manifest/crypto_test.go:16]: Remember that we use table driven tests by default (but no need to change this now).

**Review by katexochen — APPROVED** [review]

File: internal/manifest/crypto.go (hunk 1 above)
Comment (burgerdev) [reply internal/manifest/crypto.go:22]: It's called `Marshal...` and `Parse...` in the `x509` package, though. Which inconsistency would you rather have?

File: internal/manifest/crypto_test.go (hunk 2 above)
Comment (burgerdev) [reply internal/manifest/crypto_test.go:18]: I tried to use the `testCase` pattern, but eventually realized that we want the full cartesian product. The tests for wrong key and wrong label are somewhat special and don't seem to fit naturally into a test table.

**Review by 3u13r — APPROVED** [review]

**Pull request merged**